### PR TITLE
update.py: auto-generate common-partners.rst from partner_list.py before building

### PR DESCRIPTION
**Pull request by nkruzan**
I made an attempt to make managing table of partner images/links easier.

I tested the output file locally with another script, but I do not have a wiki build env setup to fully test this. (maybe outfile path is wrong)

comments or improvements welcome.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
tried this...does not seem to be generating any rst file....I removed the common-partners.rst from common subdirectory and built wiki for dev....it did not produce a file that I can see

**Comment by nkruzan** [comment]
> tried this...does not seem to be generating any rst file....I removed the common-partners.rst from common subdirectory and built wiki for dev....it did not produce a file that I can see

is possibly the path is bad here?
```generate_common_partners_page('common/source/docs/common-partners.rst')```

**Comment by nkruzan** [comment]
I test locally with this ```#!/usr/bin/env python3

import os
def generate_common_partners_page(output_file):
    import importlib.util
    script_dir = os.path.dirname(__file__)
    spec = importlib.util.spec_from_file_location("partner_list.py",os.path.join(script_dir,"partner_list.py"))
    mod = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(mod)
    partners = mod.PARTNER_LIST
    image_width = '250px'
    image_align = 'center'
    image_url_base = '../../../images/supporters/'
    table_width = 2
    cell_count = 0
    output = []
    output.append(""".. _common-partners:

==================
Corporate Partners
==================

The ArduPilot Project is greatly appreciative of our Corporate Partners.
Details on the Partners Program and how to join can be found on the :doc:`Partners Program <common-partners-program>` page.

.. list-table::
    :class: partners-table\n""")
        
    for partner in partners:
        cell_count += 1
        if cell_count == 1:
            output.append('    *')

        output.append('     - .. image:: '+image_url_base+partner.image)
        output.append('            :width: '+image_width)
        output.append('            :align: '+image_align)
        output.append('            :target: '+partner.link+'\n')
        if cell_count == table_width:
            cell_count = 0

    while cell_count > 0 and cell_count < table_width:
        output.append('     - .. ')
        cell_count += 1

    output.append('\n[copywiki destination="copter,plane,rover,planner,planner2,antennatracker,dev,ardupilot,mavproxy"]')
    
    with open(output_file, 'w') as out:
        out.write('\n'.join(output))


#generate_partners_table('common/source/docs/common-partners.rst')
generate_common_partners_page('common-partners.rst')``` and get common-partners.rst created in same directory with the following contents:
```.. _common-partners:

==================
Corporate Partners
==================

The ArduPilot Project is greatly appreciative of our Corporate Partners.
Details on the Partners Program and how to join can be found on the :doc:`Partners Program <common-partners-program>` page.

.. list-table::
    :class: partners-table

    *
     - .. image:: ../../../images/supporters/supporters_logo_EAMS.png
            :width: 250px
            :align: center
            :target: https://eams-robo.co.jp

     - .. image:: ../../../images/supporters/supporters_logo_jDrones.png
            :width: 250px
            :align: center
            :target: http://jdrones.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_CubePilot.png
            :width: 250px
            :align: center
            :target: https://www.cubepilot.com/#/home

     - .. image:: ../../../images/supporters/supporters_logo_mRobotics.png
            :width: 250px
            :align: center
            :target: https://mrobotics.io

    *
     - .. image:: ../../../images/supporters/supporters_logo_Emlid.png
            :width: 250px
            :align: center
            :target: https://emlid.com

     - .. image:: ../../../images/supporters/supporters_logo_CUAV.jpg
            :width: 250px
            :align: center
            :target: http://www.cuav.net

    *
     - .. image:: ../../../images/supporters/supporters_logo_Craft_and_Theory.png
            :width: 250px
            :align: center
            :target: http://craftandtheoryllc.com

     - .. image:: ../../../images/supporters/supporters_logo_LightWare.png
            :width: 250px
            :align: center
            :target: https://lightwarelidar.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_SpektreWorks.png
            :width: 250px
            :align: center
            :target: https://spektreworks.com

     - .. image:: ../../../images/supporters/supporters_logo_CubePilot.png
            :width: 250px
            :align: center
            :target: https://www.cubepilot.com/#/home

    *
     - .. image:: ../../../images/supporters/supporters_logo_Dronebility.png
            :width: 250px
            :align: center
            :target: http://dronebility.com

     - .. image:: ../../../images/supporters/supporters_logo_Drone_Japan.jpg
            :width: 250px
            :align: center
            :target: https://drone-j.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_RobSense.png
            :width: 250px
            :align: center
            :target: http://robsense.com

     - .. image:: ../../../images/supporters/supporters_logo_BlueRobotics.png
            :width: 250px
            :align: center
            :target: https://bluerobotics.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Skyrocket.jpg
            :width: 250px
            :align: center
            :target: https://sky-viper.com

     - .. image:: ../../../images/supporters/supporters_logo_Drotek.png
            :width: 250px
            :align: center
            :target: https://drotek.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Harris_Aerial.jpg
            :width: 250px
            :align: center
            :target: https://www.harrisaerial.com

     - .. image:: ../../../images/supporters/supporters_logo_AltiGator.jpg
            :width: 250px
            :align: center
            :target: https://altigator.com/en

    *
     - .. image:: ../../../images/supporters/supporters_logo_Event_38.png
            :width: 250px
            :align: center
            :target: https://event38.com

     - .. image:: ../../../images/supporters/supporters_logo_3DXR.jpg
            :width: 250px
            :align: center
            :target: https://3dxr.co.uk

    *
     - .. image:: ../../../images/supporters/supporters_logo_Carbonix.png
            :width: 250px
            :align: center
            :target: https://carbonix.com.au

     - .. image:: ../../../images/supporters/supporters_logo_Drones_Center.png
            :width: 250px
            :align: center
            :target: https://drones-center.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Wurzbach_Electronics.png
            :width: 250px
            :align: center
            :target: https://wurzbachelectronics.com

     - .. image:: ../../../images/supporters/supporters_logo_TT_Robotix.jpg
            :width: 250px
            :align: center
            :target: http://ttrobotix.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Air_Supply_Aerial.png
            :width: 250px
            :align: center
            :target: https://airsupply.com

     - .. image:: ../../../images/supporters/supporters_logo_IR_Lock.jpg
            :width: 250px
            :align: center
            :target: https://irlock.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Benewake.png
            :width: 250px
            :align: center
            :target: http://en.benewake.com

     - .. image:: ../../../images/supporters/supporters_logo_Foxtech.jpg
            :width: 250px
            :align: center
            :target: https://foxtechfpv.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_BFD_Systems.jpg
            :width: 250px
            :align: center
            :target: https://bfdsystems.com

     - .. image:: ../../../images/supporters/supporters_logo_Rubidium_Light.jpg
            :width: 250px
            :align: center
            :target: https://rubidiumlight.com.au/rubidium-rover

    *
     - .. image:: ../../../images/supporters/supporters_logo_Makeflyeasy.jpg
            :width: 250px
            :align: center
            :target: http://www.makeflyeasy.com

     - .. image:: ../../../images/supporters/supporters_logo_Hexsoon.jpg
            :width: 250px
            :align: center
            :target: http://www.hexsoon.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Micro_Aerial_Projects.png
            :width: 250px
            :align: center
            :target: https://microaerialprojects.com

     - .. image:: ../../../images/supporters/supporters_logo_ARACE_UAS.png
            :width: 250px
            :align: center
            :target: https://araceuas.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Qifei.png
            :width: 250px
            :align: center
            :target: https://agrobot.en.alibaba.com

     - .. image:: ../../../images/supporters/supporters_logo_Mateksys.png
            :width: 250px
            :align: center
            :target: http://www.mateksys.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Freespace.png
            :width: 250px
            :align: center
            :target: https://freespacesolutions.com.au

     - .. image:: ../../../images/supporters/supporters_logo_Holybro.png
            :width: 250px
            :align: center
            :target: http://www.holybro.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_ASW.png
            :width: 250px
            :align: center
            :target: https://aerosystemswest.com

     - .. image:: ../../../images/supporters/supporters_logo_Bask_Aerospace.jpg
            :width: 250px
            :align: center
            :target: https://baskaerospace.com.au

    *
     - .. image:: ../../../images/supporters/supporters_logo_D_Makers.png
            :width: 250px
            :align: center
            :target: https://dmakers.co.kr

     - .. image:: ../../../images/supporters/supporters_logo_Hitec.png
            :width: 250px
            :align: center
            :target: https://hitecnology.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_AeroScanTech.png
            :width: 250px
            :align: center
            :target: https://aeroscantech.com

     - .. image:: ../../../images/supporters/supporters_logo_Quaternium.png
            :width: 250px
            :align: center
            :target: https://quaternium.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_NP_UAS_TS.png
            :width: 250px
            :align: center
            :target: http://npuasts.com

     - .. image:: ../../../images/supporters/supporters_logo_AION_ROBOTICS.png
            :width: 250px
            :align: center
            :target: https://aionrobotics.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_ALTI.jpg
            :width: 250px
            :align: center
            :target: https://altiuas.com

     - .. image:: ../../../images/supporters/supporters_logo_Flytrex.png
            :width: 250px
            :align: center
            :target: https://flytrex.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_VimDrones.png
            :width: 250px
            :align: center
            :target: https://vimdrones.com

     - .. image:: ../../../images/supporters/supporters_logo_UAV_Systems_International.jpg
            :width: 250px
            :align: center
            :target: https://uavsystemsinternational.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_UAVEX.png
            :width: 250px
            :align: center
            :target: https://uavex.com

     - .. image:: ../../../images/supporters/supporters_logo_Geodrones_Australia.png
            :width: 250px
            :align: center
            :target: https://geodronesaustralia.com.au

    *
     - .. image:: ../../../images/supporters/supporters_logo_Dual_RC.png
            :width: 250px
            :align: center
            :target: https://dualrc.com

     - .. image:: ../../../images/supporters/supporters_logo_DAVWINGS.png
            :width: 250px
            :align: center
            :target: http://davwings.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_FrSky.png
            :width: 250px
            :align: center
            :target: https://frsky-rc.com

     - .. image:: ../../../images/supporters/supporters_logo_TaiwanDrone100.png
            :width: 250px
            :align: center
            :target: https://taiwandrone100.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Kraus_Aerospace.png
            :width: 250px
            :align: center
            :target: https://krausaerospace.com

     - .. image:: ../../../images/supporters/supporters_logo_Argosdyne.png
            :width: 250px
            :align: center
            :target: http://argosdyne.com/en

    *
     - .. image:: ../../../images/supporters/supporters_logo_PDRL.png
            :width: 250px
            :align: center
            :target: https://pdrl.in

     - .. image:: ../../../images/supporters/supporters_logo_RadioMaster.png
            :width: 250px
            :align: center
            :target: https://radiomasterrc.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Skydroid.png
            :width: 250px
            :align: center
            :target: http://www.skydroid.xin

     - .. image:: ../../../images/supporters/supporters_logo_Asylon.png
            :width: 250px
            :align: center
            :target: https://dronecore.us

    *
     - .. image:: ../../../images/supporters/supporters_logo_Watts_Innovation.png
            :width: 250px
            :align: center
            :target: https://wattsinnovations.com

     - .. image:: ../../../images/supporters/supporters_logo_Exyn.png
            :width: 250px
            :align: center
            :target: https://exyn.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_Telluraves_Aerospace.jpg
            :width: 250px
            :align: center
            :target: https://telluraves.co.za

     - .. image:: ../../../images/supporters/supporters_logo_Ardusimple.png
            :width: 250px
            :align: center
            :target: https://www.ardusimple.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_qiotek.jpg
            :width: 250px
            :align: center
            :target: https://www.qio-tek.com

     - .. image:: ../../../images/supporters/supporters_logo_apd.png
            :width: 250px
            :align: center
            :target: https://powerdrives.net

    *
     - .. image:: ../../../images/supporters/supporters_logo_BZB_UAS.png
            :width: 250px
            :align: center
            :target: https://bzbuas.com/

     - .. image:: ../../../images/supporters/supporters_logo_Ocius.jpg
            :width: 250px
            :align: center
            :target: https://ocius.com.au/

    *
     - .. image:: ../../../images/supporters/supporters_logo_RotorsAndCams.png
            :width: 250px
            :align: center
            :target: https://rotorsandcams.com/

     - .. image:: ../../../images/supporters/supporters_logo_blicube.jpg
            :width: 250px
            :align: center
            :target: https://www.blicube.com/

    *
     - .. image:: ../../../images/supporters/supporters_logo_ModalAI.png
            :width: 250px
            :align: center
            :target: https://www.modalai.com/

     - .. image:: ../../../images/supporters/supporters_ing_logo.png
            :width: 250px
            :align: center
            :target: https://www.ingrobotic.com/

    *
     - .. image:: ../../../images/supporters/supporters_logo_SYNEREX.png
            :width: 250px
            :align: center
            :target: http://www.synerex.kr/

     - .. image:: ../../../images/supporters/supporters_logo_Offshore_Aviation.png
            :width: 250px
            :align: center
            :target: https://www.Offshoreaviation.com

    *
     - .. image:: ../../../images/supporters/supporters_logo_currawong.png
            :width: 250px
            :align: center
            :target: https://www.currawongeng.com/

     - .. image:: ../../../images/supporters/supporters_logo_skyways.png
            :width: 250px
            :align: center
            :target: https://skyways.com/

    *
     - .. image:: ../../../images/supporters/supporters_logo_Sky-Drones.png
            :width: 250px
            :align: center
            :target: https://sky-drones.com/

     - .. image:: ../../../images/supporters/supporters_tata_logo.png
            :width: 250px
            :align: center
            :target: https://www.tataadvancedsystems.com/

    *
     - .. image:: ../../../images/supporters/supporters_logo_sparkletech.jpg
            :width: 250px
            :align: center
            :target: https://www.sparkleuav.com/

     - .. image:: ../../../images/supporters/supporters_vulcan_logo.jpg
            :width: 250px
            :align: center
            :target: https://vulcanuav.com/

    *
     - .. image:: ../../../images/supporters/supporters_fly_dragon_logo.png
            :width: 250px
            :align: center
            :target: https://www.droneassemble.com/

     - .. image:: ../../../images/supporters/supporters_horizon31_logo.png
            :width: 250px
            :align: center
            :target: https://www.horizon31.com/

    *
     - .. image:: ../../../images/supporters/supporters_skylift_logo.png
            :width: 250px
            :align: center
            :target: http://www.skylift.aero/

     - .. image:: ../../../images/supporters/supporters_logo_airvolute.png
            :width: 250px
            :align: center
            :target: https://www.airvolute.com/

    *
     - .. image:: ../../../images/supporters/supporters_hylio_logo.png
            :width: 250px
            :align: center
            :target: https://www.hyl.io

     - .. 

[copywiki destination="copter,plane,rover,planner,planner2,antennatracker,dev,ardupilot,mavproxy"]```

File: partner_list.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    Partner('supporters_logo_Sky-Drones.png','https://sky-drones.com/'),
+    Partner('supporters_tata_logo.png','https://www.tataadvancedsystems.com/'),
+    Partner('supporters_logo_sparkletech.jpg','https://www.sparkleuav.com/'),
+    Partner('supporters_vulcan_logo.jpg','https://vulcanuav.com/'),
+    Partner('supporters_fly_dragon_logo.png','https://www.droneassemble.com/'),
+    Partner('supporters_horizon31_logo.png','https://www.horizon31.com/'),
+    Partner('supporters_skylift_logo.png','http://www.skylift.aero/'),
+    Partner('supporters_logo_airvolute.png','https://www.airvolute.com/'),
+    Partner('supporters_hylio_logo.png','https://www.hyl.io'),
+]
```
[the remark this answers is not in the document]
Comment (nkruzan) [reply partner_list.py:101]: I'm not sure what you mean

[the remark this answers is not in the document]
Comment (nkruzan) [reply partner_list.py:101]: if you mean the red circle with - in it, that is github showing there is not a line feed at the end of line

**Comment by nkruzan** [comment]
@Hwurzburg  perhaps the better method is just to include this in the scripts directory and whomever updates the partner list need to manually run script to generate updated page?

**Comment by Hwurzburg** [comment]
that would work, only Rishab, Randy or I usually do that page.......as it stands, I cant get an output file from the above...maybe if its made into a standalone script generating an rst file in the directory from which its executed?

**Comment by nkruzan** [comment]
do you know is wiki build on python2 or python3?

**Comment by Hwurzburg** [comment]
I think its python3 now..  @peterbarker ?

**Comment by brunoolivieri** [comment]
The server default is python 2.7.18. I particularly, whenever I can, force to use python3.

Note that there is a docker environment and a vagrant enabled in the repository as well.

**Comment by Hwurzburg** [comment]
closing this since Rishab has it under control now

**Pull request closed without merging**